macos/FluentUI/Notification/Notification.swift

```diff
@@ -0,0 +1,241 @@
+						messageLabel
+						Spacer(minLength: 0)
+					}
+					.accessibilityElement(children: .combine)
+					button
+						.layoutPriority(1)
+				}
+				.frame(minHeight: Constants.minimumHeight)
+				.padding(.horizontal, Constants.horizontalPadding)
+				.clipped()
+			}
+		}
+
+		@ViewBuilder
+		var notification: some View {
+			innerContents
+				.background(
+					backgroundColor
+				)
+				.overlay(Rectangle().frame(width: nil, height: state.style == .subtle ? Constants.outlineWidth : 0, alignment: .top).foregroundColor(Constants.neutralBackgroundColor), alignment: state.showFromBottom ? .top : .bottom)
+				.onTapGesture {
+					if let messageAction = messageButtonAction {
+						messageAction()
+					}
+				}
+		}
+
+		return notification
+	}
+
+	var state: MSFNotificationStateImpl
+
+	var backgroundColor: Color
+
+	var foregroundColor: Color
+
+	private struct Constants {
+		static let accentBackgroundColor: Color = Color(FluentUI.Colors.primaryTint30)
+		static let accentForegroundColor: Color = Color(FluentUI.Colors.primaryShade20)
+		static let subtleBackgroundColor: Color = Color(.white)
+		static let subtleForegroundColor: Color = Color(FluentUI.Colors.primary)
+		static let neutralBackgroundColor: Color = Color(FluentUI.Colors.Palette.gray100.color)
+		static let neutralForegroundColor: Color = Color(FluentUI.Colors.Palette.gray900.color)
+
+		static let horizontalPadding: CGFloat = 16
+		static let horizontalSpacing: CGFloat = 16
+		static let minimumHeight: CGFloat = 45
+		static let outlineWidth: CGFloat = 0.5
+	}
+}
+
+class MSFNotificationStateImpl: NSObject, MSFNotificationState {
+	@Published var message: String?
+	@Published var showFromBottom: Bool
+
+	/// Title to display in the action button on the trailing edge of the control.
+	///
+	/// To show an action button, provide values for both `actionButtonTitle` and `actionButtonAction`.
+	@Published var actionButtonTitle: String?
+
+	/// Action to be dispatched by the action button on the trailing edge of the control.
+	///
+	/// To show an action button, provide values for both `actionButtonTitle` and `actionButtonAction`.
+	@Published var actionButtonAction: (() -> Void)?
+
+	/// Action to be dispatched by tapping on the toast/bar notification.
+	@Published var messageButtonAction: (() -> Void)?
+
+	/// Style to draw the control.
+	@Published var style: MSFNotificationStyle
+
+    @objc convenience init(style: MSFNotificationStyle) {
+        self.init(style: style,
+                  message: nil,
+                  actionButtonTitle: nil,
+                  actionButtonAction: nil,
+                  messageButtonAction: nil,
+                  showFromBottom: true)
+	}
+
+    init(style: MSFNotificationStyle,
+         message: String? = nil,
+         actionButtonTitle: String? = nil,
+         actionButtonAction: (() -> Void)? = nil,
+         messageButtonAction: (() -> Void)? = nil,
+         showFromBottom: Bool = true) {
```

Comment:
we should make it enum. just incase  like notiication animates from the side like system notfication